### PR TITLE
add api to retrieve user's specific onboarded ProductDetails

### PR DESCRIPTION
**Pull request by KevinSi96**
#### List of Changes

add api to retrieve user's specific onboarded ProductDetails

#### Motivation and Context

External interceptor needs to retrieve the date of the user's onboarding to the product

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: core/src/main/java/it/pagopa/selfcare/external_api/core/UserServiceImpl.java [hunk 1]
```diff
@@ -54,4 +59,36 @@ public UserInfoWrapper getUserInfo(String fiscalCode) {
         return result;
     }
 
+    @Override
+    public UserDetailsWrapper getUserOnboardedProductDetails(String userId, String institutionId, String productId) {
+        log.trace("getUserOnboardedProductDetails start");
+        log.debug("getUserOnboardedProductDetails userId = {}, institutionId = {}, productId = {}", userId, institutionId, productId);
+        UserDetailsWrapper result = null;
+        OnboardingInfoResponse onboardingInfoResponse = msCoreConnector.getInstitutionProductsInfo(userId);
```
Comment (manuraf) [inline core/src/main/java/it/pagopa/selfcare/external_api/core/UserServiceImpl.java:67]: does exist any api on ms-core that return only onboarded-product for that institution? you might avoid to get entire list and filtering here...

File: core/src/main/java/it/pagopa/selfcare/external_api/core/UserServiceImpl.java [hunk 2]
```diff
@@ -54,4 +59,36 @@ public UserInfoWrapper getUserInfo(String fiscalCode) {
...
+        Optional<OnboardedInstitutionResponse> institutionResponse = onboardingInfoResponse.getInstitutions().stream()
+                .filter(onboardedInstitutionResponse -> institutionId.equals(onboardedInstitutionResponse.getId())
+                        && productId.equals(onboardedInstitutionResponse.getProductInfo().getId()))
```
Comment (manuraf) [inline core/src/main/java/it/pagopa/selfcare/external_api/core/UserServiceImpl.java:70]: does api return only product ACTIVE? if there are more than one product you may return a wrong object

File: core/src/main/java/it/pagopa/selfcare/external_api/core/UserServiceImpl.java (hunk 1 above)
Comment (KevinSi96) [reply core/src/main/java/it/pagopa/selfcare/external_api/core/UserServiceImpl.java:67]: There arent any ready ones

File: core/src/main/java/it/pagopa/selfcare/external_api/core/UserServiceImpl.java (hunk 2 above)
Comment (manuraf) [reply core/src/main/java/it/pagopa/selfcare/external_api/core/UserServiceImpl.java:70]: yes, for example if there are a product DELETED and another product ACTIVE the first will be the DELETED one

Comment (KevinSi96) [reply core/src/main/java/it/pagopa/selfcare/external_api/core/UserServiceImpl.java:70]: ah Ok i get it. I just checked and only ACTIVE and Pending Ones are retrieved by default

**Review by manuraf — APPROVED** [review]

**Pull request merged**